sdk/python/feast/infra/offline_stores/contrib/trino_offline_store/trino.py

```diff
@@ -0,0 +1,567 @@
+
+
+class TrinoRetrievalJob(RetrievalJob):
+    def __init__(
+        self,
+        query: str,
+        client: Trino,
+        config: RepoConfig,
+        full_feature_names: bool,
+        on_demand_feature_views: Optional[List[OnDemandFeatureView]] = None,
+        metadata: Optional[RetrievalMetadata] = None,
+    ):
+        self._query = query
+        self._client = client
+        self._config = config
+        self._full_feature_names = full_feature_names
+        self._on_demand_feature_views = on_demand_feature_views
+        self._metadata = metadata
+
+    @property
+    def full_feature_names(self) -> bool:
+        return self._full_feature_names
+
+    @property
+    def on_demand_feature_views(self) -> Optional[List[OnDemandFeatureView]]:
+        return self._on_demand_feature_views
+
+    def _to_df_internal(self) -> pd.DataFrame:
+        """Return dataset as Pandas DataFrame synchronously including on demand transforms"""
+        results = self._client.execute_query(query_text=self._query)
+        self.pyarrow_schema = results.pyarrow_schema
+        return results.to_dataframe()
+
+    def _to_arrow_internal(self) -> pyarrow.Table:
+        """Return payrrow dataset as synchronously including on demand transforms"""
+        return pyarrow.Table.from_pandas(
+            self._to_df_internal(), schema=self.pyarrow_schema
+        )
+
+    def to_sql(self) -> str:
+        """Returns the SQL query that will be executed in Trino to build the historical feature table"""
+        return self._query
+
+    def to_trino(
+        self,
+        destination_table: Optional[str] = None,
+        timeout: int = 1800,
+        retry_cadence: int = 10,
+    ) -> Optional[str]:
+        """
+        Triggers the execution of a historical feature retrieval query and exports the results to a Trino table.
+        Runs for a maximum amount of time specified by the timeout parameter (defaulting to 30 minutes).
+        Args:
+            timeout: An optional number of seconds for setting the time limit of the QueryJob.
+            retry_cadence: An optional number of seconds for setting how long the job should checked for completion.
+        Returns:
+            Returns the destination table name.
+        """
+        if not destination_table:
+            today = date.today().strftime("%Y%m%d")
+            rand_id = str(uuid.uuid4())[:7]
+            destination_table = f"{self._client.catalog}.{self._config.offline_store.dataset}.historical_{today}_{rand_id}"
+
+        # TODO: Implement the timeout logic
+        query = f"CREATE TABLE {destination_table} AS ({self._query})"
+        self._client.execute_query(query_text=query)
+        return destination_table
+
+    def persist(self, storage: SavedDatasetStorage):
+        """
+        Run the retrieval and persist the results in the same offline store used for read.
+        """
+        assert isinstance(storage, SavedDatasetTrinoStorage)
```

Comment:
Raise value error instead of using assert

Reply:
Yes this is best practice. We've been using `assert` in a lot of places in the code ...

Changed it